ui/components/nance/RetroactiveRewards.tsx

```diff
@@ -0,0 +1,394 @@
+          projects,
+          addressToQuadraticVotingPower
+        )
+      : {}
+
+  const { contract: distributionTableContract } = useContract(
+    DISTRIBUTION_TABLE_ADDRESSES[chain.slug]
+  )
+  const { tokens } = useAssets()
+  const { ethBudget, usdBudget, mooneyBudget, ethPrice } = getBudget(
+    tokens,
+    year,
+    quarter
+  )
+  const {
+    projectIdToETHPayout,
+    projectIdToMooneyPayout,
+    addressToEthPayout,
+    addressToMooneyPayout,
+    ethPayoutCSV,
+    mooneyPayoutCSV,
+  } = getPayouts(
+    projectIdToEstimatedPercentage,
+    projects,
+    ethBudget,
+    mooneyBudget
+  )
+
+  const handleSubmit = async () => {
+    const totalPercentage = Object.values(distribution).reduce(
+      (sum, value) => sum + value,
+      0
+    )
+    if (totalPercentage !== 100) {
+      toast.error('Total distribution must equal 100%', {
+        style: toastStyle,
+      })
+      return
+    }
+    // Fill in empty values with 0
+    // Missing values are assumed to be projects which the user is a contributor to
+    // and will be filled in with iterative normalization
+    for (const project of projects) {
+      if (
+        !(project.id in distribution) &&
+        !(
+          !userAddress ||
+          userAddress in project.contributors ||
+          userAddress.toLowerCase() in project.contributors
+        )
+      ) {
+        distribution[project.id] = 0
+      }
+    }
+    try {
+      if (edit) {
+        await distributionTableContract?.call('updateTableCol', [
+          quarter,
+          year,
+          JSON.stringify(distribution),
+        ])
+        toast.success('Distribution edited successfully!', {
+          style: toastStyle,
+        })
+        setTimeout(() => {
+          refreshRewards()
+        }, 5000)
+      } else {
+        await distributionTableContract?.call('insertIntoTable', [
+          quarter,
+          year,
+          JSON.stringify(distribution),
+        ])
+        toast.success('Distribution submitted successfully!', {
+          style: toastStyle,
+        })
+        setTimeout(() => {
+          refreshRewards()
+        }, 5000)
+      }
+    } catch (error) {
+      console.error('Error submitting distribution:', error)
+      toast.error('Error submitting distribution. Please try again.', {
+        style: toastStyle,
+      })
+    }
+  }
+  const handleDelete = async () => {
+    try {
+      await distributionTableContract?.call('deleteFromTable', [quarter, year])
+      toast.success('Distribution deleted successfully!', {
+        style: toastStyle,
+      })
+      setTimeout(() => {
+        refreshRewards()
+      }, 5000)
+    } catch (error) {
+      console.error('Error deleting distribution:', error)
+      toast.error('Error deleting distribution. Please try again.', {
+        style: toastStyle,
+      })
+    }
+  }
+
+  return (
+    <section id="projects-container" className="overflow-hidden">
+      <Head title="Projects" image="" />
```

Comment:
title="Retroactive Rewards" description="Distribute rewards to contributors based on their contributions."

dont define the image prop